### ISSUE TITLE
Got "hostname appears in more than one automation policy, making certificate management ambiguous" after updating to v2.3.0-beta.1

### ISSUE THREAD

**Issue by Lycreal** [caddyserver/caddy#3906]
## 1. Caddy version

```
v2.3.0-beta.1 h1:AcPy8kAnN6LsxBVtW6aeYBFdiRW8Sz4aMwc3tUKyefc=
```

### 2a. What happens

I want handle certificates with caddy but I dont want 80->8443 redirects. `caddy adapt` fails for Caddyfile below 

```
a.a {
  tls internal
  respond 403
}

http://b.b https://b.b:8443 {
  tls internal
  respond 404
}
```

But same Caddyfile works fine for caddy v2.2.1. I see similar issue #3618, but this should have been fix in v2.2.0.
So it's strange to see this happen again in v2.3.0-beta.1. I guess this is a bug introduced in v2.3.0-beta.1.

### 2b. Log output

```
$ caddy adapt
2020/12/07 10:54:13.670 INFO    using adjacent Caddyfile
adapt: hostname appears in more than one automation policy, making certificate management ambiguous: b.b
```

### 2c. Workaround(s)

```
a.a {
  tls internal
  respond 403
}

http://b.b {
  respond 404
}

https://b.b:8443 {
  tls internal
  respond 404
}
```
Below Caddyfile works for v2.3.0-beta.1 too. it's obviously abnormal because there's no large differences between the Caddyfile below and the Caddyfile on the top.
```
http://b.b https://b.b:8443 {
  tls internal
  respond 404
}
```

**Comment by mholt** [comment]
Thank you for the great bug report! I appreciate that it was simplified down to the minimal required reproduce case. I was able to experience the behavior you described quite easily and have a fix coming in a bit.

**Comment by mholt** [comment]
Fixed in 7e71915 - again, I really appreciate the great report and for you testing the beta!

**Comment by Lycreal** [comment]
Glad to be of help 😀
